### DISCUSSION TITLE
Travel time for buses

### DISCUSSION THREAD

**Discussion by blaurent49** [valhalla/valhalla#4900] [Q&A] [answered]
Hello,

I have installed Valhalla via Docker. I would like to compute routes for buses. For the moment, I am not happy with the result I get: the travel times are too fast. 

From the client side, I tried to add the following code in my resquests but to no avail:

```
"costing_options":
{"bus":
    {
    "maneuver_penalty": 30,
     "turn_penalty": 0,
      "service_penalty": 30,
      "traffic_signal_penalty": 30,
      "speed_types": {
        "highway": 100,
        "trunk": 90,
        "primary": 80,
        "secondary": 70,
        "tertiary": 60,
        "residential": 50,
        "service": 20,
        "motorway": 100,
        "motorway_link": 60,
        "trunk_link": 60,
        "primary_link": 50,
        "secondary_link": 40,
        "tertiary_link": 30,
        "living_street": 20,
        "unclassified": 10,
        "track": 10
      }
    }
}
```

I also tried adding to the same code in the valhalla.son file (server-side) but with no more effect.

The only result I got is with the `top_speed` option in my requests but I find it too "brutal".

I am fairly new to Valhalla. Is possible to reduce speed limits for buses according the type of lane? If yes, how can I do?

Thank you in advance for your help.

**Comment by chrstnbwnkl** [comment]
Getting ETA's right is difficult. The simplest way to go is to get historical traffic data and use that to enrich the graph. This can be quite expensive, so depending on your budget, there's an alternative: Valhalla is able to set edge speeds from a JSON, depending on edge road class and admin region (see "default_speeds_config" in the docs). The default JSON Valhalla uses for this is derived from millions of map matched GPS traces, so it's as close as you can get in terms of historical traffic without busting the bank.

Another hint I can give you is that the bus profile uses pretty much the same configurations as the auto profile, maybe you'd want to tweak things like transition costs a bit more to your liking (this is not configurable though, so you'd have to do that in code).

**Reply by blaurent49** [reply]
Thank you for you answer.

As the name "default_speeds_config" suggests, the default config is already loaded by default in Valhalla? Do I have something special to do? I beg your pardon but I did find "default_speeds_config" in the docs.

**Reply by chrstnbwnkl** [reply]
It depends on your setup. If you are using [this image](https://github.com/gis-ops/docker-valhalla), then yes, you don't need to do anything as it loads the JSON by default.

**Reply by blaurent49** [reply]
This is indeed the image I use but unfortunately, I am not satisfied with the results I get with my container :-(.

I tried to compute a route between two locations on this api: https://valhalla.openstreetmap.de/directions?profile=bus&wps=5.0194715%2C47.3702632%2C4.7327154%2C47.6655703

![image](https://github.com/user-attachments/assets/3c888dc4-82cb-4dab-82a9-467166fd7557)

The result I get is better than the one obtained with my own container. So, there must exist a better setting than mine...

**Reply by chrstnbwnkl** [reply, marked as answer]
Yeah, if you are wondering about differences between the default docker setup from our GIS-OPS image vs. the public instance, there's a discussion here: #4894. Basically, our admins are broken on the public instance so the default speed assignment is not working as expected. If you're happier with the public instances ETA's you can try to disable the default speed config :smile:

**Reply by blaurent49** [reply]
> you can try to disable the default speed config
How can I do that? (sorry, I am beginning with Valhalla)

**Reply by chrstnbwnkl** [reply]
Just check the docs of the docker image you're using

**Reply by blaurent49** [reply]
Ok, I got it: 

> use_default_speeds_config: True loads a JSON file used to enhance default speeds (or falls back to an existing custom_files/default_speeds.json) and sets the respective config entry. Read more [here](https://github.com/OpenStreetMapSpeeds/schema). Default False

According to the Docs, the gis-ops container does not use the default_speeds.json file by default. 

I downloaded it in my custom_files/ folder and set the `use_default_speeds_config` parameter to True but again with no effect. I can't see nothing related to this config file in the logs. I do not know if it is properly loaded or not...

**Reply by blaurent49** [reply]
I eventually forced the rebuild and got some changes in the results. Here is the command line I used:

`docker run -dt --name valhalla_gis-ops -e use_default_speeds_config=True -e force_rebuild=True -p 8002:8002 -v ${PWD}:/custom_files ghcr.io/gis-ops/docker-valhalla/valhalla:latest `

Now I need to evaluate the results I'm getting in more depth. 

Anyway, thank you for your help.
